django/contrib/admin/sites.py

```diff
@@ -588,6 +590,27 @@ def app_index(self, request, app_label, extra_context=None):
             context,
         )
 
+    @cached_property
+    def site_content_types(self):
+        from django.contrib.contenttypes.models import ContentType
+
+        content_types = []
+        for model in self._registry.keys():
+            content_types.append(
+                ContentType.objects.get(
+                    app_label=model._meta.app_label, model=model._meta.model_name
+                )
+            )
+        return content_types
+
+    def get_log_entries(self):
+        from django.contrib.admin.models import LogEntry
+
+        entries = LogEntry.objects.select_related("content_type", "user")
+        if self.log_registered_models:
+            entries = entries.filter(content_type__in=self.site_content_types)
+        return entries
```

Comment:
I think I'd call this `log_entries_queryset`, and the same in key in the `each_context` method. 🤔

We should just provide the base implementation. `return LogEntry.objects.select_related("content_type", "user")`

Do we need to nest the import? 🤔

Reply:
thanks @carltongibson for reviewing.

> Do we need to nest the import?

yes, otherwise we have a circular import.

> I think I'd call this `log_entries_queryset`, and the same in key in the `each_context` method.

IMO, `log_entries_queryset` doesn't sound right. It either shall be plural, ie. `log_entries`, or a queryset of instances of `LogEntry`-s, hence `log_entry_queryset`. If that's a merge stopper, I'll of course fix it.